hazelcast/src/main/java/com/hazelcast/config/MemberDomConfigProcessor.java

```diff
@@ -110,7 +111,7 @@
 
     private static final ILogger LOGGER = Logger.getLogger(MemberDomConfigProcessor.class);
 
-    private final Config config;
+    protected final Config config;
```

Comment:
We need `MemberDomConfigProcessor` because `AbstractDomConfigProcessor` is extended by `ClientDomConfigProcessor` too. Also, `MemberDomConfigProcessor` currently contains the whole XML parsing logic and the parts shared between XML and YAML logic. It is possible to move the parts specific to XML and overridden in the `YamlMemberDomConfigProcessor` to an `XmlMemberDomConfigProcessor`. I actualy agree that it would be cleaner. But, I would postpone this change to after the other PRs touching member side config are merged.

Like the idea, keeping it in mind :+1:

Reply:
Ok, sounds good. I didn't notice `ClientDomConfigProcessor`. Then, you could even have a structure like:

```
                                                                   <- XmlMemberDomConfigProcessor
AbstractDomConfigProcessor <- AbstractMemberDomConfigProcessor
                                                                   <- YamMemberDomConfigProcessor
```

Anyway, I'm totally fine to change it later in a separate PR.